app/controllers/workspaces_controller.rb

```diff
@@ -76,8 +76,36 @@ def destroy
     end
   end
 
+  # POST /workspaces/1/favorite
+  def favorite
+    params.permit(:favorite, :id)
+    authorize! :favorite, @workspace
+
+    if params[:favorite]
+      @workspace.favorited_by << current_user
+    else
+      @workspace.favorited_by.delete(current_user)
+    end
+
+    respond_to do |format|
+      if @workspace.save
+        format.html { redirect_to @workspace, notice: favorite_notice(params[:favorite]) }
```

Comment:
This notice is never seen in practice because all the requests come through as JSON requests.